scripts/migrations/tournaments/main.go

```diff
@@ -0,0 +1,516 @@
+	DivisionID   string `gorm:"uniqueIndex:idx_registrant"`
+}
+
+
+func NewDBStore(config *config.Config, gs gameplay.GameStore) (*DBStore, error) {
+	db, err := gorm.Open(postgres.Open(config.DBConnString), &gorm.Config{})
+	if err != nil {
+		return nil, err
+	}
+	db.AutoMigrate(&tournament{})
+	db.AutoMigrate(&registrant{})
+	return &DBStore{db: db, gameStore: gs, cfg: config}, nil
+}
+
+func (s *DBStore) ListAllIDs(ctx context.Context) ([]string, error) {
+	var tids []struct{ UUID string }
+	ctxDB := s.db.WithContext(ctx)
+
+	result := ctxDB.Table("tournaments").Select("uuid").Order("created_at").Scan(&tids)
+	ids := make([]string, len(tids))
+	for idx, tid := range tids {
+		ids[idx] = tid.UUID
+	}
+	return ids, result.Error
+}
+
+func (s *DBStore) Set(ctx context.Context, tm *entity.Tournament) error {
+	dbt, err := s.toDBObj(tm)
+	if err != nil {
+		return err
+	}
+
+	ctxDB := s.db.WithContext(ctx)
+	result := ctxDB.Model(&tournament{}).Clauses(clause.Locking{Strength: "UPDATE"}).
+		Where("uuid = ?", tm.UUID).Updates(dbt)
+
+	return result.Error
+}
+
+func (s *DBStore) toDBObj(t *entity.Tournament) (*tournament, error) {
+
+	directors, err := json.Marshal(t.Directors)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, division := range t.Divisions {
+		dmJSON, err := json.Marshal(division.DivisionManager)
+		if err != nil {
+			return nil, err
+		}
+
+		division.DivisionRawMessage = dmJSON
+	}
+
+	divisions, err := json.Marshal(t.Divisions)
+	if err != nil {
+		return nil, err
+	}
+
+	defaultSettings, err := json.Marshal(t.DefaultSettings)
+	if err != nil {
+		// for now
+		defaultSettings = []byte("{}")
+	}
+
+	dbt := &tournament{
+		UUID:              t.UUID,
+		Name:              t.Name,
+		Description:       t.Description,
+		AliasOf:           t.AliasOf,
+		Directors:         directors,
+		ExecutiveDirector: t.ExecutiveDirector,
+		IsStarted:         t.IsStarted,
+		IsFinished:        t.IsFinished,
+		Divisions:         divisions,
+		DefaultSettings:   defaultSettings,
+		Type:              string(t.Type),
+		Parent:            t.ParentID,
+		Slug:              t.Slug,
+	}
+	return dbt, nil
+}
+
+// Main
+
+func main() {
+	// Populate every game with its quickdata
+	cfg := &config.Config{}
+	cfg.Load(os.Args[1:])
+	log.Info().Msgf("Loaded config: %v", cfg)
+
+	zerolog.SetGlobalLevel(zerolog.InfoLevel)
+	
+	userStore, err := user.NewDBStore(cfg.DBConnString)
+	if err != nil {
+		panic(err)
+	}
+
+	tmpGameStore, err := game.NewDBStore(cfg, userStore)
+	if err != nil {
+		panic(err)
+	}
+	gameStore := game.NewCache(tmpGameStore)
+
+	tournamentStore, err := NewDBStore(cfg, gameStore)
+	if err != nil {
+		panic(err)
+	}
+
+	ctx := context.Background()
+
+	ids, err := tournamentStore.ListAllIDs(ctx)
+	if err != nil {
+		panic(err)
+	}
+	log.Info().Interface("ids", ids).Msg("listed-game-ids")
+
+	for _, tid := range ids {
+		_, err := oldDatabaseObjectToEntity(ctx, tournamentStore, tid)
+		if err != nil {
+			panic(err)
```

Comment:
is it possible to wrap this entire thing in a transaction? a panic in the middle is scary

Reply:
Yes.